dipy/segment/tests/test_metric.py

```diff
@@ -174,7 +174,7 @@ def dist(self, v1, v2):
             if metric.are_compatible(f1.shape, f2.shape):
                 distance = metric.dist(f1, f2)
                 if np.all(f1 == f2):
-                    assert_equal(distance, 0.)
+                    assert_almost_equal(distance, 0.)
```

Comment:
I did a debug build using `try_branch.py` : http://nipy.bic.berkeley.edu/builders/dipy-py2.6-32/builds/553/steps/shell_6/logs/stdio

The error comes out of the Cython code, yes.  I can't reproduce it with the same code in numpy, I think the same precision.  Still not sure where it comes from.